tests/unit/graph_learning_test.py

```diff
@@ -1661,18 +1727,36 @@ def test_5lm_feature_experiment(self):
         config = copy.deepcopy(self.feature_5lm_config)
         with MontyObjectRecognitionExperiment(config) as exp:
             pprint("...training...")
-            exp.train()
-
-            train_stats = pd.read_csv(
-                os.path.join(exp.output_dir, "train_stats.csv")
-            )
-            # The following check is brittle and depends on sensor arrangement. Leaving
-            # the rest of the test intact to detect run failures, but disabling checking
-            # of particular results.
-            # self.check_multilm_train_results(train_stats, num_lms=5, min_done=3)
+            objects = [self.fake_obs_learn, self.fake_obs_house_3d]
+            graph_modules = self.get_5lm_gm_with_fake_objects(objects)
 
             pprint("...evaluating...")
-            exp.evaluate()
+            monty = exp.model
+            monty.set_experiment_mode("eval")
+            monty.learning_modules = [gm.learning_module for gm in graph_modules]
+
+            for gm in graph_modules:
+                gm.mode = "eval"
+                gm.pre_episode(self.placeholder_target)
+
+            exp.pre_epoch()
+            # Use the first graph module's observations for the episodes
+            # since it has no offsets like the others do.
+            for episode in graph_modules[0].episodes:
+                exp.pre_episode()
+                monty.pre_episode(self.placeholder_target)
+                for observation in episode.observations:
```

Comment:
does this have to have the same name as when we extract observations on line 1752? To me this would be more intuitive to be `for step in range(episode.observations)` (kind of how I did it on my branch on line 1728 and 1732 https://github.com/thousandbrainsproject/tbp.monty/compare/main...vkakerbeck:tbp.monty:my_5lm_test_branch)

Reply:
I am probably missing something about what type `observation` has here and how you extract the observations at each step on the next line. But I am confused where the `step` comes in in general. Like where do you specify which step in the experiment we are in to not keep extracting the first observation each time.

Reply:
Yep, your right, I messed this up during the refactor of our original rough code to the state it's in now. Let me clean it up and see if it makes more sense then.